src/Mod/PartDesign/Gui/TaskPadParameters.ui

```diff
@@ -40,124 +40,165 @@
      </property>
-     <property name="checkable">
-      <bool>true</bool>
-     </property>
-     <layout class="QGridLayout" name="gridLayout_2">
+     <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
-       <widget class="QLabel" name="labelXSkew">
-        <property name="text">
-         <string>x</string>
-        </property>
-       </widget>
-      </item>
-      <item row="0" column="1">
-       <widget class="Gui::DoubleSpinBox" name="XDirectionEdit">
-        <property name="toolTip">
-         <string>x-component of direction vector</string>
-        </property>
-        <property name="keyboardTracking">
-         <bool>false</bool>
-        </property>
-        <property name="minimum">
-         <double>-100.000000000000000</double>
-        </property>
-        <property name="maximum">
-         <double>100.000000000000000</double>
-        </property>
-        <property name="singleStep">
-         <double>0.100000000000000</double>
-        </property>
-        <property name="unit" stdset="0">
-         <string notr="true"/>
-        </property>
-       </widget>
+       <layout class="QGridLayout" name="gridLayout_3">
+        <item row="0" column="0">
+         <widget class="QLabel" name="labelEdge">
+          <property name="text">
+           <string>Direction/edge:</string>
+          </property>
+         </widget>
+        </item>
+        <item row="0" column="1">
+         <widget class="QComboBox" name="directionCB">
+          <property name="toolTip">
+           <string>Either along the sketch normal
+or select an edge as reference</string>
+          </property>
+          <item>
+           <property name="text">
+            <string>Sketch normal</string>
+           </property>
+          </item>
+          <item>
+           <property name="text">
+            <string>Select reference...</string>
+           </property>
+          </item>
+         </widget>
+        </item>
+       </layout>
       </item>
       <item row="1" column="0">
-       <widget class="QLabel" name="labelYSkew">
-        <property name="text">
-         <string>y</string>
+       <widget class="QGroupBox" name="groupBoxDirection">
+        <property name="enabled">
+         <bool>true</bool>
         </property>
-       </widget>
-      </item>
-      <item row="1" column="1">
-       <widget class="Gui::DoubleSpinBox" name="YDirectionEdit">
         <property name="toolTip">
-         <string>y-component of direction vector</string>
-        </property>
-        <property name="keyboardTracking">
-         <bool>false</bool>
-        </property>
-        <property name="minimum">
-         <double>-100.000000000000000</double>
-        </property>
-        <property name="maximum">
-         <double>100.000000000000000</double>
-        </property>
-        <property name="singleStep">
-         <double>0.100000000000000</double>
+         <string>Use custom vector for pad direction otherwise
```

Comment:
I'd put the phrase "otherwise the sketch plane's normal vector will be used" in parenthesis, or put a comma or colon before it.

Reply:
done